lib/Doctrine/ODM/MongoDB/Configuration.php

```diff
@@ -347,7 +348,7 @@ public function getClassMetadataFactoryName(): string
 
     public function getDefaultCommitOptions(): array
     {
-        return $this->attributes['defaultCommitOptions'] ?? ['w' => 1];
+        return $this->attributes['defaultCommitOptions'] ?? ['writeConcern' => new WriteConcern(1)];
```

Comment:
Even though a write concern is a lightweight object, I feel like we don't need to instantiate it every time:

```suggestion
        if (! isset($this->attributes['defaultCommitOptions'])) {
            $this->attributes['defaultCommitOptions'] = ['writeConcern' => new WriteConcern(1)];
        }

        return $this->attributes['defaultCommitOptions'];
```

Reply:
Actually, right hand side of this operator is not executed unless the isset test that is part of this operator return false.

You can check it in the following script:

```php
<?php
declare(strict_types=1);

class Test {
    public function __construct()
    {
        die('x');
    }
}

$test = 1;

$test2 = $test ?? new Test();

echo 'done';

// output will be "done"
```

Reply:
What I'm saying is that if you repeatedly call `getDefaultCommitOptions` without any options set, it will always return a new instance of `WriteConcern` with the next call once again instantiating a new object since the attributes array was not modified.